src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -500,8 +522,11 @@ public final boolean supports(SCMSourceDescriptor descriptor) {
          * @throws InterruptedException if the attempt to create a {@link SCMFileSystem} was interrupted.
          */
         @CheckForNull
-        public abstract SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull SCMRevision rev)
-                throws IOException, InterruptedException;
+        public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull SCMRevision rev)
+                throws IOException, InterruptedException
+        {
+            throw new AbstractMethodError("Implement build(Item, SCM, SCMRevision,Run<?,?>)");
```

Comment:
```suggestion
            throw new AbstractMethodError("Implement build(Item, SCM, SCMRevision, Run)");
```

Reply:
Hmm but that wouldnt match the signature?  . And I made the signature that way because I saw it also in GitSCM.checkout

Reply:
Generic types are not part of the method signature at the bytecode level.